### PR TITLE
Changes needed when running v4

### PR DESCRIPTION
**Pull request by klaricch**

### CODE REVIEW

File: gnomad_constraint/resources/resource_utils.py [hunk 1]
```diff
@@ -125,7 +125,13 @@ def get_methylation_ht(build: str) -> TableResource:
...
         return ref_grch37.methylation_sites
     elif build == "GRCh38":
-        return ref_grch38.methylation_sites
+        methylation_chrx = methylation_sites_chrx.ht()
+        methylation_autosomes = methylation_sites.ht()
+        methylation_ht = methylation_autosomes.union(methylation_chrx)
+        methylation_ht = methylation_ht.checkpoint("gs://gnomad-tmp/methylation_ht.ht")
+        return TableResource(
+            path="gs://gnomad-tmp/methylation_ht.ht",
+        )
```
Comment (jkgoodrich) [inline gnomad_constraint/resources/resource_utils.py:134]: Maybe this?

```suggestion
        tmp_path = get_constraint_root(version=build, test=True)
        methylation_ht = methylation_ht.checkpoint(tmp_path, overwrite=True)
        return TableResource(path=tmp_path)
```

File: gnomad_constraint/resources/resource_utils.py [hunk 2]
```diff
@@ -125,7 +125,13 @@ def get_methylation_ht(build: str) -> TableResource:
     if build == "GRCh37":
         return ref_grch37.methylation_sites
     elif build == "GRCh38":
-        return ref_grch38.methylation_sites
+        methylation_chrx = methylation_sites_chrx.ht()
+        methylation_autosomes = methylation_sites.ht()
```
Comment (jkgoodrich) [inline gnomad_constraint/resources/resource_utils.py:129]:
```suggestion
        methylation_chrx = ref_grch38.methylation_sites_chrx.ht()
        methylation_autosomes = ref_grch38.methylation_sites.ht()
```

File: gnomad_constraint/utils/constraint.py [hunk 3]
```diff
@@ -112,21 +112,28 @@ def prepare_ht_for_constraint_calculations(
     # collapse strands to deduplicate the context.
     ht = annotate_mutation_type(collapse_strand(ht))
 
+    # Obtain field name for median exome coverage.
```
Comment (jkgoodrich) [inline gnomad_constraint/utils/constraint.py:115]: I know this is going to change anyway for v4, but as is it also won't work with the stratified coverage HT

File: gnomad_constraint/pipeline/constraint_pipeline.py [hunk 4]
```diff
@@ -484,6 +491,8 @@ def main(args):
                 },
                 min_diff_convergence=args.min_diff_convergence,
                 raw_z_outlier_threshold=args.raw_z_outlier_threshold,
+                include_os=int(version[0])
+                < 4,  # OS (other splice) is not implemented for build 38
```
Comment (jkgoodrich) [inline gnomad_constraint/pipeline/constraint_pipeline.py:495]:
```suggestion
                < 4,  # OS (other splice) is not implemented for build 38.
```

File: gnomad_constraint/utils/constraint.py [hunk 5]
```diff
@@ -730,6 +738,7 @@ def compute_constraint_metrics(
         variants), whereas values either above '--raw-z-outlier-threshold' or below
         the negative of '--raw-z-outlier-threshold' will be considered outliers for
         synonymous varaint counts (indicating too few or too many variants).
+    :param include_os: Whether or not to include OS (other splice) as a grouping when stratifiying calculations by lof HC.
```
Comment (jkgoodrich) [inline gnomad_constraint/utils/constraint.py:741]:
```suggestion
    :param include_os: Whether or not to include OS (other splice) as a grouping when 
        stratifying calculations by lof HC.
```

**Review by jkgoodrich — CHANGES_REQUESTED** [review]
A couple suggestions

File: gnomad_constraint/utils/constraint.py (hunk 3 above)
Comment (klaricch) [reply gnomad_constraint/utils/constraint.py:115]: added todo

**Review by jkgoodrich — APPROVED** [review]
LGTM!

**Pull request merged**